lib/faraday/adapter/test.rb

```diff
@@ -160,6 +160,12 @@ def matches?(stack, env)
           end
           nil
         end
+
+        # Executes the given block in a mutex context to avoid multi-thread race conditions
+        def stubs_mutex(&block)
+          @stubs_mutex ||= Monitor.new
```

Comment:
We could create this one in the initializer, to always have it on hand, knowing it would be created.

Reply:
And actually, avoid race conditions on its creation 😄! Good point 🙌